frontends/rioterm/src/main.rs

```diff
@@ -179,7 +179,30 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         }
 
         if let Some(working_dir_cli) = args.window_options.terminal_options.working_dir {
-            config.working_dir = Some(working_dir_cli);
+            let wd_canonical = match std::fs::canonicalize(&working_dir_cli).and_then(
+                |path| {
+                    if path.is_dir() {
+                        path.into_os_string().into_string().map_err(|_| {
+                            std::io::Error::new(
+                                std::io::ErrorKind::InvalidData,
+                                "Invalid UTF-8 in path",
+                            )
+                        })
+                    } else {
+                        Err(std::io::Error::new(
+                            std::io::ErrorKind::NotADirectory,
+                            "Path is not a directory",
+                        ))
+                    }
+                },
+            ) {
+                Ok(canonical_path) => Some(canonical_path),
```

Comment:
one more suggestion, let's get rid of the variable, something like:

match std::fs::canonicalize(&working_dir_cli)
    .and_then(|path| {
        if path.is_dir() {
            path.into_os_string().into_string().map_err(|_| 
                std::io::Error::new(std::io::ErrorKind::InvalidData, "Invalid UTF-8 in path"))
        } else {
            Err(std::io::Error::new(std::io::ErrorKind::NotADirectory, "Path is not a directory"))
        }
    })
{
    Ok(canonical_path) => {
        config.current_working_directory = canonical_path;
    },
    Err(e) => {
        tracing::warn!("Failed to set working directory '{}': {}. Using default instead.", working_dir_cli, e);
        None
    }
};